gammapy/estimators/utils.py

```diff
@@ -157,3 +155,71 @@ def estimate_exposure_reco_energy(dataset, spectral_model=None, normalize=True):
         reco_exposure = reco_exposure / ref_flux[:, np.newaxis, np.newaxis]
 
     return reco_exposure
+
+
+def resample_energy_edges(dataset, conditions={}):
+    """Returns energy edges that defining a binning that satisfies given conditions on the per-bin statistics.
+
+    Parameters
+    ----------
+    dataset:`~gammapy.datasets.SpectrumDataset` or `~gammapy.datasets.SpectrumDatasetOnOff`
+            the input dataset
+    conditions : dict
+        Keyword arguments containing the per-bin conditions used to resample the axis.
+        Available options are: 'counts_min', 'background_min', 'excess_min', 'sqrt_ts_min', 'npred_min',
+        'npred_background_min', 'npred_signal_min'.
+    Returns
+    -------
+    energy_edges : list of `~astropy.units.Quantity`
+        Energy edges for the resampled energy axis.
+    """
+    if not isinstance(dataset, SpectrumDataset) or isinstance(
+        dataset, SpectrumDatasetOnOff
+    ):
+        raise NotImplementedError(
+            "This method is currently supported for spectral datasets only."
+        )
+
+    available_conditions = [
+        "counts_min",
+        "background_min",
+        "excess_min",
+        "sqrt_ts_min",
+        "npred_min",
+        "npred_background_min",
+        "npred_signal_min",
+    ]
+    for key in conditions.keys():
+        if key not in available_conditions:
+            raise ValueError(
+                f"Unrecognized option {key}. The available methods are: {available_conditions}."
+            )
+
+    def _satisfies_conditions(info_dict):
```

Comment:
I think this could be moved outside and just take `conditions` as well. This make the code structure probably a bit simpler...

Reply:
@adonath I did move it outside but didn't really get how not to pass it the `info_dict`

Reply:
What you did is exactly what I meant!